linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala

```diff
@@ -0,0 +1,202 @@
+ */
+
+package org.apache.linkis.manager.engineplugin.hbase.executor
+
+import org.apache.linkis.common.conf.Configuration
+import org.apache.linkis.common.utils.{OverloadUtils, Utils}
+import org.apache.linkis.engineconn.computation.executor.execute.{
+  ConcurrentComputationExecutor,
+  EngineExecutionContext
+}
+import org.apache.linkis.engineconn.core.EngineConnObject
+import org.apache.linkis.governance.common.protocol.conf.{
+  RequestQueryEngineConfig,
+  ResponseQueryConfig
+}
+import org.apache.linkis.manager.common.entity.resource.{
+  CommonNodeResource,
+  LoadResource,
+  NodeResource
+}
+import org.apache.linkis.manager.engineplugin.common.util.NodeResourceUtils
+import org.apache.linkis.manager.engineplugin.hbase.conf.HBaseConfiguration
+import org.apache.linkis.manager.engineplugin.hbase.constant.HBaseEngineConnConstant
+import org.apache.linkis.manager.engineplugin.hbase.shell.{
+  HBaseShellSession,
+  HBaseShellSessionManager,
+  Result
+}
+import org.apache.linkis.manager.label.entity.Label
+import org.apache.linkis.manager.label.entity.engine.{EngineTypeLabel, UserCreatorLabel}
+import org.apache.linkis.protocol.CacheableProtocol
+import org.apache.linkis.protocol.engine.JobProgressInfo
+import org.apache.linkis.rpc.{RPCMapCache, Sender}
+import org.apache.linkis.scheduler.executer.{
+  ErrorExecuteResponse,
+  ExecuteResponse,
+  SuccessExecuteResponse
+}
+
+import org.apache.commons.collections.CollectionUtils
+
+import java.util
+import java.util.Collections
+
+import scala.collection.JavaConverters._
+
+class HBaseEngineConnExecutor(val id: Int) extends ConcurrentComputationExecutor {
+  private val shellSessionManager = HBaseShellSessionManager.getInstance();
+  private val executorLabels: util.List[Label[_]] = new util.ArrayList[Label[_]](2)
+
+  override def init(): Unit = {
+    logger.info("hbase executor start init.")
+    super.init()
+  }
+
+  override def executeLine(
+      engineExecutorContext: EngineExecutionContext,
+      code: String
+  ): ExecuteResponse = {
+    val realCode = code.trim()
+    var properties: util.Map[String, String] = Collections.emptyMap()
+    Utils.tryCatch({
+      properties = getHBaseRuntimeParams(engineExecutorContext)
+    }) { e: Throwable =>
+      logger.error(s"try to build hbase runtime params error! $e")
+      return ErrorExecuteResponse(e.getMessage, e)
+    }
+    logger.info(s"The hbase properties is: $properties")
+    var shellSession: HBaseShellSession = null
+    Utils.tryCatch({
+      shellSession = shellSessionManager.getHBaseShellSession(properties)
+    }) { e: Throwable =>
+      logger.error(s"created hbase shell session error! $e")
+      return ErrorExecuteResponse("created hbase shell session error!", e)
+    }
+    val result: Result = shellSession.execute(realCode)
+    if (!result.isSuccess) {
+      return ErrorExecuteResponse(result.getResult, result.getThrowable)
+    }
+    engineExecutorContext.appendStdout(result.getResult)
```

Comment:
Is it necessary to add a result set？

Reply:
The execution result of HBase Shell is a bunch of strings. Does it need to be packaged into a result set?

Reply:
If  not table we need to write text result set

Reply:
<img width="684" alt="image" src="https://github.com/apache/linkis/assets/27730211/41f0caf9-bdc5-4510-bbf1-22c84a295116">
ResultSetFactory.TEXT_TYPE?